aten/src/TH/THGenerator.hpp

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <ATen/core/MT19937RNGEngine.h>
+
+/**
+ * THGeneratorState is a POD class needed for memcpys
+ * in torch.get_rng_state() and torch.set_rng_state().
+ * It is a legacy class and even though it is replaced with
+ * at::CPUGenerator, we need this class and some of its fields
+ * to support backward compatibility on loading checkpoints.
```

Comment:
I'm actually not sure we need to maintain BC here, but since you've done so, I'll take it. We can always break BC later.

Reply:
Oh, I see, you've implemented a limited form of BC, where you can load old checkpoints, but saved checkpoints are incompatible with new versions.

Reply:
Yes, I wanted to support loading of old checkpoints in set_rng_state, however, get_rng_state would return a THGeneratorNew as a byte tensor.